### ISSUE TITLE
I'm interested in why it performs better

### ISSUE THREAD

**Issue by shuoshuoa** [FateScript/CenterNet-better#1]

**Comment by FateScript** [comment]
> _No description provided._

It might be caused by the following reasons:
* I fix some bugs in original repo, for example, torchvision pretrained model should use specific mean provided in document; torchvision model require RGB image while opencv is BGR and so on. I fixed lots of detail setting. 
* My repo use DDP while the original repo use DP, this might case slight difference.
* I use torch.clamp and set eps to 1e-12 to make loss function more stable
* Different version of Pytorch
* I use a different optimizer setting(However, I tried adam and it also has comparable performance).

**Comment by shuoshuoa** [comment]
Thanks for your reply, I got it. By the way, you can try TTFNet's strategy, which is similar to CenterNet

**Comment by developer0hye** [comment]
Hi @FateScript ,

Thanks for sharing your work!

Why can clamping sigmoid value help a model be trained more stably?

**Comment by FateScript** [comment]
@developer0hye `log` operator in loss might cause NaN value, clipping it ensures numberic stablity.

**Comment by developer0hye** [comment]
@FateScript thanks!

**Comment by developer0hye** [comment]
@FateScript 

Actually, I am trying to reproduce centernetv1 from scratch. I could achieve comparable performance on VOC dataset. However, I couldn't achieve it on COCO(my implementation:27.61ap, paper:28.1ap, resnet18).

https://github.com/MarkAny-Vision-AI/CenterNeXt/issues/13

I am doing extensive experiments to reproduce on COCO dataset.

Can you give me some advice...??

I found that my model detects large objects well but has low ap on small objects.

**Comment by FateScript** [comment]
@developer0hye I'm not sure what you did with your code. Many things might cause  lower performance. You could use aug/model/optimizer in `CenterNet-better` to check your componet, it might cause some gpu-hours.

**Comment by developer0hye** [comment]
@FateScript 
Thanks for your advice!
